src/application/shell/command/Mathexpr.cpp

```diff
@@ -0,0 +1,156 @@
+            while(i < tokens.length() && tokens[i]>=48 && tokens[i]<=57){
+                val = (val*10) + (tokens[i]-'0');
+                i++;
+            }
+             
+            values.add(val);
+            i--;
+        }
+         
+        /* If closing bracket is encountered then solve the entire brace */ 
+        else if(tokens[i] == ')') {
+            while(!ops.isEmpty() && ops.get(ops.size()-1) != '(') {
+                int val2 = values.get(values.size()-1);
+                values.remove(values.size()-1);
+                 
+                int val1 = values.get(values.size()-1);
+                values.remove(values.size()-1);
+                 
+                char op = ops.get(ops.size()-1);
+                ops.remove(ops.size()-1);
+                 
+                values.add(applyOperation(val1, val2, op));
+            }
+            
+            /* Remove the opening brace from the list */
+            if(!ops.isEmpty()) ops.remove(ops.size()-1);
+        }
+         
+        /*Else the current token is the operator */
+        else {
+            /* On basis of the precedence and other factors calculate the expression */
+            while(!ops.isEmpty() && precedence(ops.get(ops.size()-1)) >= precedence(tokens[i])){
+                int val2 = values.get(values.size()-1);
+                values.remove(values.size()-1);
+                 
+                int val1 = values.get(values.size()-1);
+                values.remove(values.size()-1);
+                 
+                int op = ops.get(ops.size()-1);
+                ops.remove(ops.size()-1);
+                 
+                values.add(applyOperation(val1, val2, op));
+            }
+             
+            
+            ops.add(tokens[i]);
+        }
+    }
+     
+    /* The expression has been calculated, now we only need to calculate the remaining ops to the values */
+    while(!ops.isEmpty()){
+        int val2 = values.get(values.size()-1);
+        values.remove(values.size()-1);
+                 
+        int val1 = values.get(values.size()-1);
+        values.remove(values.size()-1);
+                 
+        int op = ops.get(ops.size()-1);
+        ops.remove(ops.size()-1);
+                 
+        values.add(applyOperation(val1, val2, op));
+    }
+    
+    /* Returning the evaluted value of the expression */
+    return values.get(values.size()-1);
+}
+
+void Mathexpr::execute(Util::Array<String> &args) {
+    Util::ArgumentParser parser(getHelpText(), 1);
+
+    /* Parser checks for the error if any on the basis of the command entered by the user */
+    if(!parser.parse(args)) {
+        stderr << args[0] << ": " << parser.getErrorString() << endl;
+        return;
+    }
+    /* Check whether more than one unamed arguments are given by the user and display the appropriate message it they do */
+    if(parser.getUnnamedArguments().length()>1){
+        stderr << args[0] << ": Invalid command" << endl;
```

Comment:
Could you please give a more detailed error message (e.g. "Invalid number of arguments")?